### ISSUE TITLE
References to class methods and properties in changelogs

### ISSUE THREAD

**Issue by pondermatic** [gocodebox/lifterlms#1912]
**TL;DR**: Should `::` be used in changelog files (not DocBlocks) even on non-static methods and properties?

There are two categories of changelogs.
1. the `@see` tag in PHPDoc DocBlocks
2. files in `/.changelogs/` or the `CHANGELOG.md` file

Which operator should references to class methods and properties in changelogs use?
1. use the static [Scope Resolution Operator `::`](https://www.php.net/manual/en/language.oop5.paamayim-nekudotayim.php#language.oop5.paamayim-nekudotayim) regardless of whether the item is static or not
1. use `::` for static items and `->` for non-static items

In PHPDoc, `::` is always used. See the FQSEN definition in [PSR-5](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#3-definitions). Therefore, in order to provide automatic linking to other elements, references in DocBlocks should always use the `::` operator.

However, there is not a standard for references in changelog files. LifterLMS uses `::` more frequently than `->`.

My preference is to use the operator that the calling code would use. For example:
- `LLMS_Example::add_error()` for a static method
- `LLMS_Example->display_errors()` for a non-static method
- `LLMS_Example::errors` for a static property
- `LLMS_Example->errors` for a non static property

When searching for a function or property name that may also show up in strings or comments, I often prefix the search term with `->` for non-static access or `::` for static access. If the changelog files always use `::`, then non-static items may be missed.

It looks "weird" to me referencing a non-static method or property with the `::` operator.

In the end, it doesn't really matter to me which way we make the standard. I'm just trying to be consistent.

**Comment by thomasplevy** [comment]
@pondermatic 

I'm almost certainly entirely to blame for inconsistency. I've never taken the time to determine what the best approach is. In general I think I loosely follow always using `::` because it seems that's what WordPress does in the codex, see non-static methods: https://developer.wordpress.org/reference/classes/wp_query/get/

I say that we follow the PHPDoc spec and always use `::`. I agree that this looks weird for non-static items but I don't actually see the downside... if you know it's consistent you don't have to search for non-static items at all and just always lookup the consistent reference. Especially knowing this is the defined spec of PHPDoc I think that's what we should follow.
